backtrace-library/src/main/java/backtraceio/library/metrics/UniqueEvent.java

```diff
@@ -0,0 +1,52 @@
+package backtraceio.library.metrics;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import backtraceio.library.logger.BacktraceLogger;
+
+public class UniqueEvent extends Event {
+
+    private final static transient String LOG_TAG = UniqueEvent.class.getSimpleName();
+
+    @SerializedName("unique")
+    private List<String> name;
+
+    UniqueEvent(String name) {
+        this(name, System.currentTimeMillis() / 1000, new HashMap<String, Object>());
+    }
+
+    UniqueEvent(String name, long timestamp, Map<String, Object> attributes) {
+        super(timestamp);
+        this.attributes = attributes;
+        this.name = new ArrayList<String>() {{
+            add(name);
+        }};
+    }
+
+    // The spec specifies name as a JSON array but we don't change the interface yet
+    @Override
+    public String getName() {
+        if (this.name != null && this.name.size() > 0) {
+            return this.name.get(0);
+        }
+        BacktraceLogger.e(LOG_TAG, "Unique Event name must not be null or empty");
+        return new String();
+    }
+
+    void update(long timestamp, Map<String, Object> attributes) {
+        this.timestamp = timestamp;
+        if (attributes != null) {
+            for (String key : attributes.keySet()) {
```

Comment:
There must be utility methods for this, no? Something in Collections that is already available on the class path?

Reply:
General (I see this a lot in this code) isn't that a lib that does isNullOrEmpty instead of `value != null && !value.toString().trim().isEmpty()`

Reply:
We would need to import an additional library. I don't think @BartoszLitwiniuk would want us to add an entire new library just for `isNullOrEmpty`.

Reply:
@vlussenburg we can create our own helper function for this, good idea thank you!